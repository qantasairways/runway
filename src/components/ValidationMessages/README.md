```js
<ValidationMessages
  messageType="info"
  validationMessage="Please select a destination."
>
  Default Example Usage
</ValidationMessages>
```
