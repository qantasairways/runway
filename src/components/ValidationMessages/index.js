import React from 'react';
import { PropTypes } from 'prop-types';

import { fontSize, fontFamily, colours } from '../../theme/airways';

const MESSAGE_TYPE = {
  INFO: 'info',
  WARNING: 'warning'
};

const warningMessage = ({ messageType }) => ({
  width: '100%',
  minHeight: '48px',
  border: 'solid 0.5px rgba(151, 151, 151, 0)',
  position: 'relative',
  fontFamily: fontFamily.main,
  fontSize: fontSize.label,
  fontWeight: 'normal',
  fontStyle: 'normal',
  fontStretch: 'normal',
  letterSpacing: 'normal',
  color: colours.darkerGrey,
  padding: '11px 13px 11px 15px',
  ...(messageType === MESSAGE_TYPE.WARNING
    ? { backgroundColor: '#fcebcd' }
    : { backgroundColor: '#bff4f2' })
});

const warningArrow = () => ({
  width: 0,
  height: 0,
  borderLeft: '7px solid transparent',
  borderRight: '7px solid transparent',
  borderBottom: '7px solid #fcebcd',
  marginLeft: '15px'
});

function ValidationMessages({ validationMessage, messageType }) {
  return (
    <div role="alert">
      {messageType === MESSAGE_TYPE.WARNING && <div css={warningArrow} />}
      <div css={warningMessage({ messageType })}>{validationMessage}</div>
    </div>
  );
}

ValidationMessages.propTypes = {
  validationMessage: PropTypes.string,
  messageType: PropTypes.string
};

ValidationMessages.defaultProps = {
  validationMessage: '',
  messageType: MESSAGE_TYPE.WARNING
};

export default ValidationMessages;
