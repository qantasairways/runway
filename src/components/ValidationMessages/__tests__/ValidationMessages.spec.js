import React from 'react';
import { mount } from 'enzyme';
import ValidationMessages from '..';

describe('ValidationMessages', () => {
  it('info renders correctly', () => {
    const component = mount(
      <ValidationMessages
        messageType="info"
        validationMessage="Passengers has been reset to default value."
      />
    );

    expect(component).toMatchSnapshot();
  });

  it('warning message renders correctly', () => {
    const component = mount(
      <ValidationMessages
        messageType="warning"
        validationMessage="Please select a destination."
      />
    );

    expect(component).toMatchSnapshot();
  });
});
