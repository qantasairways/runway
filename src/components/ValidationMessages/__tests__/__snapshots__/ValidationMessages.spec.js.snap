// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ValidationMessages info renders correctly 1`] = `
.emotion-0 {
  width: 100%;
  min-height: 48px;
  border: solid 0.5px rgba(151,151,151,0);
  position: relative;
  font-family: "Ciutadella",sans-serif;
  font-size: 1rem;
  font-weight: normal;
  font-style: normal;
  font-stretch: normal;
  -webkit-letter-spacing: normal;
  -moz-letter-spacing: normal;
  -ms-letter-spacing: normal;
  letter-spacing: normal;
  color: #323232;
  padding: 11px 13px 11px 15px;
  background-color: #bff4f2;
}

<ValidationMessages
  messageType="info"
  validationMessage="Passengers has been reset to default value."
>
  <div
    role="alert"
  >
    <div
      className="emotion-0"
    >
      Passengers has been reset to default value.
    </div>
  </div>
</ValidationMessages>
`;

exports[`ValidationMessages warning message renders correctly 1`] = `
.emotion-0 {
  width: 0;
  height: 0;
  border-left: 7px solid transparent;
  border-right: 7px solid transparent;
  border-bottom: 7px solid #fcebcd;
  margin-left: 15px;
}

.emotion-1 {
  width: 100%;
  min-height: 48px;
  border: solid 0.5px rgba(151,151,151,0);
  position: relative;
  font-family: "Ciutadella",sans-serif;
  font-size: 1rem;
  font-weight: normal;
  font-style: normal;
  font-stretch: normal;
  -webkit-letter-spacing: normal;
  -moz-letter-spacing: normal;
  -ms-letter-spacing: normal;
  letter-spacing: normal;
  color: #323232;
  padding: 11px 13px 11px 15px;
  background-color: #fcebcd;
}

<ValidationMessages
  messageType="warning"
  validationMessage="Please select a destination."
>
  <div
    role="alert"
  >
    <div
      className="emotion-0"
    />
    <div
      className="emotion-1"
    >
      Please select a destination.
    </div>
  </div>
</ValidationMessages>
`;
